src/AspNetCoreModuleV2/CommonLib/FileOutputManager.cpp

```diff
@@ -32,132 +35,91 @@ FileOutputManager::Initialize(PCWSTR pwzStdOutLogFileName, PCWSTR pwzApplication
 
+    // Get the module name and add it to the log file name
+    // This change was made 
+    WCHAR path[MAX_PATH];
+    RETURN_LAST_ERROR_IF(!GetModuleFileName(g_hModule, path, sizeof(path)));
+    std::filesystem::path fsPath(path);
+    const auto moduleName = fsPath.filename().stem().c_str();
+
     GetSystemTime(&systemTime);
 
     RETURN_IF_FAILED(
-        m_struLogFilePath.SafeSnwprintf(L"%s_%d%02d%02d%02d%02d%02d_%d.log",
-        struPath.QueryStr(),
-        systemTime.wYear,
-        systemTime.wMonth,
-        systemTime.wDay,
-        systemTime.wHour,
-        systemTime.wMinute,
-        systemTime.wSecond,
-        GetCurrentProcessId()));
-
-    m_fdPreviousStdOut = _dup(_fileno(stdout));
-    m_fdPreviousStdErr = _dup(_fileno(stderr));
+        m_struLogFilePath.SafeSnwprintf(L"%s_%d%02d%02d%02d%02d%02d_%d_%s.log",
+            struPath.QueryStr(),
+            systemTime.wYear,
+            systemTime.wMonth,
+            systemTime.wDay,
+            systemTime.wHour,
+            systemTime.wMinute,
+            systemTime.wSecond,
+            GetCurrentProcessId(),
+            moduleName));
+
+    saAttr.nLength = sizeof(SECURITY_ATTRIBUTES);
+    saAttr.bInheritHandle = TRUE;
+    saAttr.lpSecurityDescriptor = NULL;
 
     m_hLogFileHandle = CreateFileW(m_struLogFilePath.QueryStr(),
         FILE_READ_DATA | FILE_WRITE_DATA,
         FILE_SHARE_READ,
         &saAttr,
         CREATE_ALWAYS,
         FILE_ATTRIBUTE_NORMAL,
-        NULL);
+        nullptr);
 
     if (m_hLogFileHandle == INVALID_HANDLE_VALUE)
     {
         return LOG_IF_FAILED(HRESULT_FROM_WIN32(GetLastError()));
     }
 
-    // There are a few options for redirecting stdout/stderr,
-    // but there are issues with most of them.
-    // AllocConsole()
-    // *stdout = *m_pStdFile;
-    // *stderr = *m_pStdFile;
-    // Calling _dup2 on stderr fails on IIS. IIS sets stderr to -2
-    // _dup2(_fileno(m_pStdFile), _fileno(stdout));
-    // _dup2(_fileno(m_pStdFile), _fileno(stderr));
-    // If we were okay setting stdout and stderr to separate files, we could use:
-    // _wfreopen_s(&m_pStdFile, struLogFileName.QueryStr(), L"w+", stdout);
-    // _wfreopen_s(&m_pStdFile, struLogFileName.QueryStr(), L"w+", stderr);
-    // Calling SetStdHandle works for redirecting managed logs, however you cannot
-    // capture native logs (including hostfxr failures).
+    stderrWrapper = std::make_unique<StdWrapper>(stderr, STD_ERROR_HANDLE, m_hLogFileHandle);
+    stdoutWrapper = std::make_unique<StdWrapper>(stdout, STD_OUTPUT_HANDLE, m_hLogFileHandle);
 
-    RETURN_LAST_ERROR_IF(!SetStdHandle(STD_OUTPUT_HANDLE, m_hLogFileHandle));
-
-    RETURN_LAST_ERROR_IF(!SetStdHandle(STD_ERROR_HANDLE, m_hLogFileHandle));
+    stderrWrapper->StartRedirection();
+    stdoutWrapper->StartRedirection();
 
     // Periodically flush the log content to file
     m_Timer.InitializeTimer(STTIMER::TimerCallback, &m_struLogFilePath, 3000, 3000);
```

Comment:
Why do we need a timer? Aren't we redirecting directly to the file?

Reply:
I agree. It was there originally for out of proc. It doesn't even look like it is flushing the file.